tests/tests.cc

```diff
@@ -0,0 +1,1112 @@
+/***  MAIN FILE WITH THE ACTUAL TESTS ***/
```

Comment:
Can we split this file into multiple files, one per logical test?

Reply:
Ok I reorganized the tests by section (with one section per file) under a new folder called tests/test_cases. Having one file per test was a little unwieldly, but now we have 5 files.